src/guide/3-behaviour-definition/1-defining-functions.rst

```diff
@@ -1,2 +1,153 @@
 Defining Agent Functions
 ========================
+
+What is an Agent Function?
+--------------------------
+
+Most behaviours within your model will be implemented using agent functions. Agent functions allow you to modify agent variables, change state
+and interact with the environment or other agents. Each agent function in FLAMEGPU2 is associated with a particular agent type and is represented
+by an ``AgentFunctionDescription`` object. This object describes the name of the function, the agent state in which it should be active, and any
+state transitions it should apply to the agent. The implementation code for the behaviour is written separately and is identified by using the
+``FLAMEGPU_AGENT_FUNCTION`` macro to label the code. Alternatively, the implementations can be written in strings and loaded at runtime. The latter
```

Comment:
would loaded be better replaced with compiled? Or did you choose loaded because non-technical users would understand it better?

Reply:
Yes, in general I am trying to avoid making things technical where possible - given the aim of FGPU2 is to hide the complexity it seems like a good default.